src/scores_ui.cpp

```diff
@@ -23,28 +26,120 @@
 #include "ui.h"
 #include "ui_manager.h"
 
-static std::string get_achievements_text( const achievements_tracker &achievements,
-        bool use_conducts, int width )
+enum class scores_ui_tab_enum : int {
+    achievements = 0,
+    conducts,
+    scores,
+    kills,
+    num_tabs
+};
+
+static scores_ui_tab_enum &operator++( scores_ui_tab_enum &c )
 {
-    // Load past game info beforehand because otherwise it may erase an `achievement_tracker`
-    // within a call to its method when lazy-loaded, causing dangling pointer.
-    get_past_games();
-    std::string thing_name = use_conducts ? _( "conducts" ) : _( "achievements" );
-    std::string cap_thing_name = use_conducts ? _( "Conducts" ) : _( "Achievements" );
+    c = static_cast<scores_ui_tab_enum>( static_cast<int>( c ) + 1 );
+    if( c == scores_ui_tab_enum::num_tabs ) {
+        c = static_cast<scores_ui_tab_enum>( 0 );
+    }
+    return c;
+}
+
+static scores_ui_tab_enum &operator--( scores_ui_tab_enum &c )
+{
+    if( c == static_cast<scores_ui_tab_enum>( 0 ) ) {
+        c = scores_ui_tab_enum::num_tabs;
+    }
+    c = static_cast<scores_ui_tab_enum>( static_cast<int>( c ) - 1 );
+    return c;
+}
+
+class scores_ui;
+
+class scores_ui
+{
+        friend class scores_ui_impl;
+    public:
+        void draw_scores_ui();
+};
+
+class scores_ui_impl : public cataimgui::window
+{
+    public:
+        std::string last_action;
+        explicit scores_ui_impl() : cataimgui::window( _( "Your scores" ),
+                    ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoNav ) {}
+
+    private:
+        void draw_achievements_text( bool use_conducts = false );
+        void draw_scores_text();
+        void draw_kills_text();
+
+        scores_ui_tab_enum selected_tab = scores_ui_tab_enum::achievements;
+        scores_ui_tab_enum switch_tab = scores_ui_tab_enum::num_tabs;
+
+        size_t window_width = str_width_to_pixels( TERMX / 3 * 2 );
+        size_t window_height = str_height_to_pixels( TERMY / 3 * 2 );
```

Comment:
Prefer something like `ImGui::GetMainViewport()->Size / 3.0 * 2.0`, since it doesn’t matter how big the monospaced font is or how many characters fit on the screen.